docs/workflow/testing/using-corerun-and-coreroot.md

```diff
@@ -0,0 +1,142 @@
+# Using Corerun To Run a .NET Application
+
+* [Introduction](#introduction)
+* [The CoreRun](#the-corerun)
+  * [Running Apps with CoreRun](#running-apps-with-corerun)
+    * [Using CoreRun with the class library from the shared system-wide .NET installation](#using-corerun-with-the-class-library-from-the-shared-system-wide-net-installation)
+    * [Using CoreRun to Execute a Published Self-Contained Application](#using-corerun-to-execute-a-published-self-contained-application)
+* [The Core_Root](#the-core_root)
+  * [Additional CoreRun Options](#additional-corerun-options)
+
+This guide will walk you through using the Corerun and Core_Root your own build from the runtime repo for testing, running apps, and so on. This doc assumes you've already built at least the _clr_ subset of the repo, and have the binaries under `artifacts/bin/coreclr/<OS>.<arch>.<configuration>`. If this is not your case, the [CoreCLR building docs](/docs/workflow/building/coreclr/README.md) have detailed instructions on how to get these artifacts.
+
+## Introduction
+
+To run a .NET app with the runtime you've built, you will need a _host_ program that will load the runtime, as well as all the other .NET libraries that your application might need. There are three main ways to go about this:
+
+* Use your machine's installed .NET SDK and replace the necessary binaries in a self-contained app.
+* Use your build's _Dev Shipping Packages_ to run your app.
+* Use the _CoreRun_ host generated as part of your build's artifacts.
+
+This guide focuses on the third of the bullet points described above. For the other two, we have docs dedicated to them:
+
+* [Using your build with your machine's installed SDK](using-your-build-with-installed-sdk.md)
+* [Using your build's dev shipping packages](using-dev-shipping-packages.md)
+
+## The CoreRun
+
+The `corerun` binary is designed to be a platform agnostic tool for quick testing of a locally built .NET runtime. In other words, this means it's able to feasibly exercise any scenario the official `dotnet` binary is capable of. It consequently helps facilitate .NET runtime development and investigation of test failures. This method is the most recommended one when you are making lots of changes that you want to keep continually testing and debugging, since it's the fastest way to apply them.
```

Comment:
I don't think that <code>corerun</code> *is able to feasibly exercise any scenario the official dotnet binary is capable of* - in fact I don't even think it's worth comparing corerun to dotnet, it's just a hosting app, its purpose is to execute tests that are easy to debug under VS or windbg. It is indeed very helpful for .NET runtime development and investigation of a certain class of test failures.

Reply:
I agree. There are many scenarios that `dotnet` supports but `corerun` does not.

corerun is a custom host for low-level runtime testing. The unique feature of corerun are special testing hooks that would not be appropriate to have in the mainstream `dotnet` host.

Reply:
> This method is the most recommended one when you are making lots of changes that you want to keep continually testing and debugging

FWIW, the standard dotnet host would work equally well in this scenario.